src/Client.php

```diff
@@ -72,7 +72,7 @@ private function createRequest(RequestInterface $request)
 
         $options['version'] = $request->getProtocolVersion();
         $options['headers'] = $request->getHeaders();
-        $options['body'] = (string) $request->getBody();
+        $options['body'] = '' === (string) $request->getBody() ? null : (string) $request->getBody();
```

Comment:
Indeed, it is easier to read. Fix, squash and add a changelog item. Using a remote url in test is not a problem ?